pkg/operator/status.go

```diff
@@ -0,0 +1,61 @@
+package operator
+
+import (
+	"fmt"
+
+	"github.com/openshift/cluster-version-operator/lib/resourceapply"
+	"github.com/openshift/cluster-version-operator/pkg/apis/clusterversion.openshift.io/v1"
+	"github.com/openshift/cluster-version-operator/pkg/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// syncStatus applies the new condition to the mco's OperatorStatus object.
+func (optr *Operator) syncStatus(cond v1.OperatorStatusCondition) error {
+	if cond.Type == v1.OperatorStatusConditionTypeDegraded {
+		return fmt.Errorf("invalid cond %s", cond.Type)
+	}
+
+	// TODO(yifan): Fill in the Extention field for the status
```

Comment:
The `Extension` field is only meant to provide extra details for the components.